src/internal/types.ts

```diff
@@ -66,7 +66,11 @@ export interface Unsubscribable {
   unsubscribe(): void;
 }
 
-export type TeardownLogic = Subscription | Unsubscribable | (() => void) | void;
+export type FinalizerLogic = Subscription | Unsubscribable | (() => void) | void;
```

Comment:
The core team have to bike shed this one a bit. 

My gut tells me that `Finalizer` is enough. But I'd need to look at usage. It's really the observable's init return that this was created for.

That's one sort of problem with a big-bang PR like this.